lens/util/repo.go

```diff
@@ -340,3 +346,118 @@ func MakeGetActorCodeFunc(ctx context.Context, store adt.Store, next, current *t
 		return act.Code, true
 	}, nil
 }
+
+// wrapper type for overloading json marshal methods
+type paramWraperType struct {
+	obj     interface{}
+	replace map[reflect.Type]func(interface{}) interface{}
+}
+
+func (wt *paramWraperType) MarshalJSON() ([]byte, error) {
```

Comment:
It would be a lot clearer if you made this a first class function:

```Go
type marshaler func(interface{}) ([]byte, error)

func MarshalWithOverrides(v interface{}, overrides map[reflect.Type]marshaler) []byte, error {
  ...
}
```
The function then uses paramWraperType as an internal implementation detail.

The marshaler returns an error so you can replace the panics with error returns in your bitfield marshaler. 

Then you can call it like this:

```Go
var bitfieldCountMarshaler = func(v interface{}) ([]byte, error) {
   ...
}

b, err := MarshalWithOverrides(b, map[reflect.Type]marshaler{
   reflect.TypeOf(bitfield.BitField{}): bitfieldCountMarshaler,
})


```